crucible-mir/src/Mir/Trans.hs

```diff
@@ -1247,11 +1250,24 @@ mkTraitObject traitName vtableName e = do
             ["vtable signature mismatch for vtable", show vtableName,
                 "of trait", show traitName, ":", show vtableTy, "!=", show vtableTy']
 
-    return $ buildTuple
+    return $ buildStructExp
         [ e
         , packAny vtable
         ]
 
+    where
+        buildStructExp :: [MirExp s] -> MirExp s
```

Comment:
Unlike the similarly named `buildStruct'` function, this does _not_ wrap each field in a `MaybeRepr`. Is it worth leaving a comment about why that is not done here?

Reply:
Renamed `buildStructExp` and added a comment to avoid confusion with `TransTy.buildStruct`: de3ff7218f54e9f04d94088d095fe63c760aca0a